app/grandchallenge/core/utils/markdown.py

```diff
@@ -9,22 +12,64 @@ def extendMarkdown(self, md):  # noqa: N802
 
 
 class BS4Treeprocessor(Treeprocessor):
+
     def run(self, root):
+
+        el_class_dict = {
+            "img": "img-fluid",
+            "blockquote": "blockquote",
+            "table": "table table-hover table-borderless",
+            "thead": "thead-light",
+            "code": "codehilite",
+        }
+
         for el in root.iter():
-            if el.tag == "img":
-                el.set("class", "img-fluid")
 
-            elif el.tag == "blockquote":
-                el.set("class", "blockquote")
+            if el.tag in el_class_dict:
+                BS4Treeprocessor.set_css_class(el, el_class_dict[el.tag])
+
+        for i in range(len(self.md.htmlStash.rawHtmlBlocks)):
+
+            bs4block = BeautifulSoup(
+                self.md.htmlStash.rawHtmlBlocks[i], "html.parser"
+            )
+
+            el = bs4block.find()
+
+            if el and el.name in el_class_dict:
+                BS4Treeprocessor.set_css_class(el, el_class_dict[el.name])
+                self.md.htmlStash.rawHtmlBlocks[i] = str(bs4block)
+
+    @staticmethod
+    def set_css_class(element, class_name: str):
+
+        if isinstance(element, ElementTree.Element):
+
+            current_class = element.attrib.get("class", "")
+
+            if class_name not in current_class:
+                new_class = f"{current_class} {class_name}".strip()
+                element.set("class", new_class)
+
+        elif isinstance(element, BeautifulSoup().new_tag("").__class__):
```

Comment:
Why are you initialising BeautifulSoup, creating a new tag, then getting the class? Why doesn't this work:

```suggestion
from bs4 import Tag
...
        elif isinstance(element, Tag):
```

Reply:
Done